### DISCUSSION TITLE
Enhancement: Support VS Code Pre-Releases

### DISCUSSION THREAD

**Discussion by nikolay** [estruyf/vscode-front-matter#889] [General]
You have two separate versions, which is not good. For quite some time already, VS Code has supported pre-releases, which comes with many benefits such as a consolidated installations number, ratings, etc.

**Comment by estruyf** [comment]
I know, but the biggest issue with pre-releases is that they need to follow sem versioning. Sem version pre-release tags on the versioning are not supported like in **npm**. 

So, if you want to prepare for a `10.5.0` release, you must release all your "pre-releases` on `10.4.x`, or follow the suggested strategy from VS Code: `major.EVEN_NUMBER.patch` for release versions and `major.ODD_NUMBER.patch for pre-release versions. 

Both, in my opinion, are not great for a good sem versioning strategy as I want to bundle multiple changes in a new release version (if it is not a bug fix) and keep a clean changelog.

That is why there is a beta version: to keep the main version following a good sem versioning strategy and have a changelog file that can follow each version release.
